x-pack/platform/plugins/shared/security/server/session_management/session_index.ts

```diff
@@ -492,8 +497,33 @@ export class SessionIndex {
         indexNeedsRefresh = (await this.bulkDeleteSessions(operations)) || indexNeedsRefresh;
       }
     } catch (err) {
-      logger.error(`Failed to clean up sessions: ${err.message}`);
-      error = err;
+      if (err instanceof errors.ResponseError && err.statusCode === 503) {
```

Comment:
As discussed, it would be good to look for `no_shard_available_action_exception` in the error message, just in case the 503 isn't related to this.